### DISCUSSION TITLE
Unable to get TCA9548A to work

### DISCUSSION THREAD

**Discussion by npant333** [zephyrproject-rtos/zephyr#45742] [General]
I am trying to use TCA9548A using zephyr. The following code returns error that "Device is not ready". I have connected the hardware properly i.e. SDA/SCL/GND/Vcc are connected properly between my processor and TCA9548A

Any help in what is not correct here?

const struct device *i2c_dev = device_get_binding(DEVICE_DT_GET(DT_NODELABEL(ch0)));
if (!device_is_ready(i2c_dev)) {
	printk("Device is not ready.\n");
	return;
}

The overlay file is as follows

&i2c0 {
	status = "okay";
	mux: tca9548a@70 {
		compatible = "ti,tca9548a";
		reg = <0x70>;
		status = "okay";
		label = "i2c_mux";
		#address-cells = <1>;
		#size-cells = <0>;

		ch0: mux_i2c@0 {
			compatible = "ti,tca9548a-channel";
			label = "mux_dw_0";
			reg = <0>;
			#address-cells = <1>;
			#size-cells = <0>;
		};

		ch1: mux_i2c@1 {
			compatible = "ti,tca9548a-channel";
			label = "mux_dw_1";
			reg = <1>;
			#address-cells = <1>;
			#size-cells = <0>;
        };
	};
};

**Comment by carlocaione** [comment]
```
const struct device *i2c_dev = DEVICE_DT_GET(DT_NODELABEL(ch0));
```

**Reply by npant333** [reply]
Thanks a lot for your suggestion. But the code is still giving the same error.

I also tried teh suggestion given in 
#https://github.com/zephyrproject-rtos/zephyr/issues/37786
CONFIG_I2C_TCA954X_ROOT_INIT_PRIO=61 
CONFIG_I2C_TCA954X_CHANNEL_INIT_PRIO=62


I also tried calling the direct i2c_read and able to get some data values. But cannot interpret them properly, like the way the TCA9548A driver code can give the values.

Many thanks for help

**Reply by zafersn** [reply]
any update @carlocaione  @npant333 ?
